src/main/java/sirius/biz/jobs/params/StringParameter.java

```diff
@@ -28,8 +30,35 @@ public StringParameter(String name, String label) {
         super(name, label);
     }
 
+    /**
+     * Marks the parameter as nullable.
+     * <p>
+     * When set, forces empty strings to be returned as {@code null} and makes it no longer required.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    public StringParameter markNullable() {
+        this.nullable = true;
+        this.required = false;
+        return self();
+    }
+
+    /**
+     * Marks the parameter as required and resets the nullable state if set.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    @Override
+    public StringParameter markRequired() {
+        this.nullable = false;
+        return super.markRequired();
+    }
+
     @Override
     protected String checkAndTransformValue(Value input) {
+        if (nullable && input.isEmptyString()) {
```

Comment:
shouldn't this be the default? null instead of "" - even ifthe parameter isn marked as nullable it still might yield null as Value.getString() does filter null - asString would do this?

Reply:
I like that! If we agree on returning null instead of "" for a zero-length input, then there is no need for a markNullable method.

Reply:
After analysis, makeNullable made obsolete and defaults to null